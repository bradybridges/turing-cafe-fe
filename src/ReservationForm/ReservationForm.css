form {
  display: flex;
  height: 40px;
  justify-content: space-around;
  margin-bottom: 50px;
  width: 90vw;
}

input {
  border-radius: 4px;
  font-size: 1em;
  text-align: center;
}

button {
  border-radius: 4px;
  font-size: 1em;
  width: 200px;
}

button:hover {
  background-color: teal;
  border: solid #000000a1 1px;
  cursor: pointer;
}