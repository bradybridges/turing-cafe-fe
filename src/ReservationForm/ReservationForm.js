import React, { Component } from 'react';
import './ReservationForm.css'

class ReservationForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      date: '',
      time: '',
      number: 0,
    }
  }

  updateReservationDetails = (e) => {
    this.setState({[e.target.name]: e.target.value});
    if(e.target.name === 'number') {
      this.setState({number: Number(e.target.value)})
    }
  }

  isFormComplete = () => {
    const isNameGood = this.state.name !== '';
    const isDateGood = this.state.date !== '';
    const isTimeGood = this.state.time !== '';
    const isNumberGood = this.state.number !== 0;

    if(isNameGood && isDateGood && isTimeGood && isNumberGood) {
      return true;
    } else {
      return false;
    }
  }

  makeReservation = (e) => {
    e.preventDefault();
    const reservation = {
      name: this.state.name,
      date: this.state.date,
      time: this.state.time,
      number: this.state.number
    };
    this.props.addReservation(reservation);
  }

  render() {
    return (
      <form>
        <input 
          onChange={this.updateReservationDetails} 
          type='text' 
          placeholder='Name' 
          name='name'
        />
        <input 
          onChange={this.updateReservationDetails} 
          type='text' 
          placeholder='Date (mm/dd)' 
          name='date'
        />
        <input 
          onChange={this.updateReservationDetails} 
          type='text' 
          placeholder='Time' 
          name='time'
        />
        <input 
          onChange={this.updateReservationDetails} 
          type='text' 
          placeholder='Number of guests'
          name='number' 
        />
        {!this.isFormComplete() && <button disabled>Make Reservation</button>}
        {this.isFormComplete() && <button onClick={this.makeReservation} >Make Reservation</button>}
      </form>
    );
  }
}

export default ReservationForm;