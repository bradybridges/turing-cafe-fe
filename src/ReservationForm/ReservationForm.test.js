import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ReservationForm from './ReservationForm';

describe('ReservationForm', () => {
  let wrapper;
  const mockAddReservation = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<ReservationForm addReservation={mockAddReservation} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('udpateReservationDetails', () => {
    it('updateReservationDetails should update the state of the component corresponding to the input that has changed', () =>{
      const mockEvent = {
        target: {name: 'name', value: 'Brady'}
      };
      wrapper.instance().updateReservationDetails(mockEvent);
      expect(wrapper.state('name')).toEqual('Brady');
    });

    it('should parse an int from input if the name of input is number and update state', () => {
      const mockEvent = {
        target: {name: 'number', value: '10'}
      }
      wrapper.instance().updateReservationDetails(mockEvent);
      expect(wrapper.state('number')).toEqual(10);
    });
  });

  it('should not make a reservation if the form is incomplete', () => {
    wrapper.instance().makeReservation = jest.fn();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().makeReservation).not.toHaveBeenCalled();
    expect(mockAddReservation).not.toHaveBeenCalled();
  });

  it('should make a reservation and call addReservation correct info when form ios complete', () => {
    const expected = {
      name: 'Brady',
      date: '10/22',
      time: '10:30',
      number: '10'
    }
    wrapper.setState(expected);
    const mockEvent = {
      preventDefault: jest.fn()
    }
    wrapper.find('button').simulate('click', mockEvent);
    expect(mockAddReservation).toHaveBeenCalledWith(expected);
  });




});