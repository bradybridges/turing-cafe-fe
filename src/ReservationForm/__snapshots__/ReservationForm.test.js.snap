// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationForm should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="text"
  />
  <button
    disabled={true}
  >
    Make Reservation
  </button>
</form>
`;
